store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -545,6 +545,20 @@ public void destroyLogics() {
         this.consumeQueueStore.destroy();
     }
 
+    private PutMessageStatus checkLmqMessage(MessageExtBrokerInner msg) {
+        if (msg.getProperties() != null
+            && StringUtils.isNotBlank(msg.getProperty(MessageConst.PROPERTY_INNER_MULTI_DISPATCH))
+            && this.isLmqConsumeQueueNumExceeded()) {
+            return PutMessageStatus.LMQ_CONSUME_QUEUE_NUM_EXCEEDED;
+        }
+        return PutMessageStatus.PUT_OK;
+    }
+
+    private boolean isLmqConsumeQueueNumExceeded() {
```

Comment:
IMO, you should split two patches:
1. add delete cq logic
2. limit queue number